Engine/Src/Ancona/Core2D/Systems/CameraSystem.cpp

```diff
@@ -42,7 +44,29 @@ void CameraComponent::MoveCamera()
                 _followPosition->position().x,
                 _followPosition->position().y);
     }
-    _view.setCenter(_view.getCenter() + _offset);
+    BoundCamera();
+    _view.setCenter(round(_view.getCenter().x), round(_view.getCenter().y));
+}
+
+void CameraComponent::BoundCamera()
+{
+    if (_view.getCenter().x > _xBounds.y)
```

Comment:
This code should be more functional. It should include the rounding and offset. You could then get rid of some of the conditions using std::max(std::min(currentY, maxY), minY);